src/Xamarin.Android.Build.Tasks/Utilities/TypeMapCecilAdapter.cs

```diff
@@ -37,11 +40,31 @@ public static (List<TypeMapDebugEntry> javaToManaged, List<TypeMapDebugEntry> ma
 
 			javaToManaged.Add (entry);
 			managedToJava.Add (entry);
+
+			if (uniqueAssemblies == null) {
+				continue;
+			}
+
+			string? asmName = td.Module.Assembly.Name.Name;
+			if (String.IsNullOrEmpty (asmName) || uniqueAssemblies.ContainsKey (asmName)) {
+				continue;
+			}
+
+			var asmInfo = new TypeMapDebugAssembly {
+				MVID = td.Module.Mvid,
+				Name = asmName,
+			};
+			asmInfo.MVIDBytes = asmInfo.MVID.ToByteArray ();
+			uniqueAssemblies.Add (asmName, asmInfo);
 		}
 
 		SyncDebugDuplicates (javaDuplicates);
 
-		return (javaToManaged, managedToJava, foundJniNativeRegistration);
+		return (new TypeMapDebugDataSets {
+			JavaToManaged = javaToManaged,
+			ManagedToJava = managedToJava,
+			UniqueAssemblies = uniqueAssemblies != null ? new List<TypeMapDebugAssembly> (uniqueAssemblies.Values) : null
```

Comment:
Should `UniqueAssemblies` be a `List<T>`? If it were e.g. `IEnumerable<TypeMapDebugAssembly>`, this could avoid an "extra" `List<TypeMapDebugAssembly>` allocation.  (Though that would also prolong the lifetime of the `Dictionary`, so maybe the `List<T>` is a better tradeoff?)

Reply:
I think `List<T>` is fine, it's lighter than the dictionary